pyfar/utils.py

```diff
@@ -0,0 +1,129 @@
+import pyfar as pf
+import numpy as np
+
+
+def broadcast_cshape(signal, cshape):
+    """
+    Broadcast signal to cshape.
+
+    The :py:mod:`cshape <pyfar._concepts.audio_classes>` of the signal is
+    broadcasted following the `numpy broadcasting rules
+    <https://numpy.org/doc/stable/user/basics.broadcasting.html>`_
+
+    Parameters
+    ----------
+    signal : Signal, TimeData, FrequencyData
+        The signal to be broadcasted.
+    cshape : tuple
+        The cshape to which `signal` is broadcasted.
+
+    Returns
+    -------
+    signal : Signal, TimeData, FrequencyData
+        The broadcasted input signal
+    """
+
+    if not isinstance(signal, (pf.Signal, pf.TimeData, pf.FrequencyData)):
+        raise TypeError("All input data must be pyfar audio objects")
+
+    signal = signal.copy()
+    signal._data = np.broadcast_to(
+        signal._data, cshape + (signal._data.shape[-1], ))
+    return signal
+
+
+def broadcast_cshapes(signals, cshape=None):
+    """
+    Broadcast multiple signals to a common cshape.
+
+    The :py:mod:`cshape <pyfar._concepts.audio_classes>` of the signals are
+    broadcasted following the `numpy broadcasting rules
+    <https://numpy.org/doc/stable/user/basics.broadcasting.html>`_
+
+    Parameters
+    ----------
+    signals : tuple of Signal, TimeData, and FrequencyData objects
```

Comment:
When building the docs locally, only _TimeData_ is recognized by sphinx and linked. Maybe we can reformulate it so all classes are linked?

Reply:
yes, linked them manually

Reply:
Locally, FrequencyData is linked now and Signal still is not... but I also think this is not very important

Reply:
Works now: https://pyfar.readthedocs.io/en/feature-bradcast_audio_classes/modules/pyfar.utils.html

Reply:
Nice! Did not work with sphinx 5.1.1 which I had installed locally, sorry!